packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -777,11 +782,26 @@ class _AppBarState extends State<AppBar> {
       _scrollNotificationObserver!.removeListener(_handleScrollNotification);
       _scrollNotificationObserver = null;
     }
+    _notificationListenerStates.clear();
     super.dispose();
   }
 
   void _handleScrollNotification(ScrollNotification notification) {
     if (notification is ScrollUpdateNotification && widget.notificationPredicate(notification)) {
+      // Each RawGestureDetector has its own scroll offset, so we need to keep track of them separately.
+      final RawGestureDetector? rawGestureDetector = notification.context?.widget as RawGestureDetector?;
+
+      if (rawGestureDetector != null) {
+        _notificationListenerStates[rawGestureDetector] = notification.metrics.axisDirection == AxisDirection.down
+        ? notification.metrics.extentBefore
+        : notification.metrics.extentAfter;
+
+        // If any of the scroll offsets are greater than 0, then the MaterialState.scrolledUnder
```

Comment:
This isn't true if they are horizontal is it?

Reply:
A dubious message in the code. 🤫